server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -237,29 +237,43 @@ public static TranslogTransferManager buildTranslogTransferManager(
 
     @Override
     public boolean ensureSynced(Location location) throws IOException {
-        try (ReleasableLock ignored = writeLock.acquire()) {
-            assert location.generation <= current.getGeneration();
-            if (location.generation == current.getGeneration()) {
-                ensureOpen();
-                return prepareAndUpload(primaryTermSupplier.getAsLong(), location.generation);
+        synchronized (uploadMutex) {
```

Comment:
can we do  uploads while taking read locks instead of using a mutex ? This is defeating the purpose of read write lock we have.

Reply:
Simplified this further, pls take a relook and we can discuss this further.